src/openrct2/interface/Screenshot.cpp

```diff
@@ -337,23 +277,47 @@ static void ReleaseDPI(rct_drawpixelinfo& dpi)
     dpi.height = 0;
 }
 
-static rct_viewport GetGiantViewport(const TileCoordsXY& mapSize, int32_t rotation, ZoomLevel zoom)
+static rct_viewport GetGiantViewport(int32_t rotation, ZoomLevel zoom)
 {
     // Get the tile coordinates of each corner
-    auto leftTileCoords = GetEdgeTile(mapSize, rotation, EdgeType::LEFT, false);
-    auto rightTileCoords = GetEdgeTile(mapSize, rotation, EdgeType::RIGHT, false);
-    auto bottomTileCoords = GetEdgeTile(mapSize, rotation, EdgeType::BOTTOM, false);
+    const TileCoordsXY cornerCoords[2][4] = {
+        {
+            // Map corners
+            { 1, 1 },
+            { gMapSize.x - 2, gMapSize.y - 2 },
+            { 1, gMapSize.y - 2 },
+            { gMapSize.x - 2, 1 },
+        },
+        {
+            // Horizontal view clipping corners
+            TileCoordsXY{ gClipSelectionA },
+            TileCoordsXY{ gClipSelectionB },
+            TileCoordsXY{ CoordsXY{ gClipSelectionA.x, gClipSelectionB.y } },
+            TileCoordsXY{ CoordsXY{ gClipSelectionB.x, gClipSelectionA.y } },
+        },
+    };
 
-    // Centre the coordinates so we don't have a hard crop at the edge of the visible tile
-    leftTileCoords += CoordsXY(16, 16);
-    rightTileCoords += CoordsXY(16, 16);
-    bottomTileCoords += CoordsXY(16, 16);
+    auto* const mainWindow = window_get_main();
+    const auto* const mainViewport = window_get_viewport(mainWindow);
+    const bool useViewClipping = (mainViewport != nullptr && mainViewport->flags & VIEWPORT_FLAG_CLIP_VIEW);
+    bool clipSelectionNotTooLarge = true;
 
+    // gMapSize must be multiplied by 32 per tile to match gClipSelection
+    if (gClipSelectionB.x - gClipSelectionA.x >= gMapSize.x * 32 || gClipSelectionB.y - gClipSelectionA.y >= gMapSize.y * 32)
+    {
+        clipSelectionNotTooLarge = false;
+    }
```

Comment:
A nicer way to handle this is to just clamp the corner coords in `corners` below to be inside gMapSize.

Reply:
See comment below